### PR TITLE
Help fix

### PR DESCRIPTION
**Pull request by anthony81799**
---
name: Help fix
about: Add help <command> and single class listing
title: 'Help fix'
lables: 'Feature update'
asignees: 'Anthony'

---

**What is the pull request for?**
Feature

**Describe what the new code does**
Allows user to list single classes and call help on individual commands.

### CODE REVIEW

File: src/main/java/controller/GUIController.java [hunk 1]
```diff
@@ -146,4 +148,43 @@ public int editMethod(String className, String oldMethod, String newMethod, Stri
 		return result;
 	}
 
+	@Override
```
Comment (ryanhaas) [inline src/main/java/controller/GUIController.java:151]: The print functions don't really need to be in the GUI controller as the GUI doesn't do much printing. It is ok for the Command and GUI controllers to be slightly different as long as the core functions are the same (which is what UMLController is for). Not a big deal, just a note.

File: src/main/java/controller/UMLController.java [hunk 2]
```diff
@@ -29,4 +31,12 @@ public UMLClassManager getModel() {
 	
 	public abstract int addRelationship(String class1, String type, String class2);
 	public abstract int removeRelationship(String class1, String type, String class2);
+
+	public abstract ArrayList<String[]> printClasses();
```
Comment (ryanhaas) [inline src/main/java/controller/UMLController.java:35]: Again, print methods likely don't really need to be in the overall controller.

File: src/main/java/model/UMLClassManager.java [hunk 3]
```diff
@@ -413,34 +466,47 @@ public int removeRelationship(String srcClass, String type, String destClass) {
...
+	/**
+	 * Lists relationships of the entire model
+	 */
+	public ArrayList<ArrayList<String[]>> printRelationships() {
+		if(relationships.isEmpty()) {
+			return null;
+		}
+		else {
+			ArrayList<ArrayList<String[]>> result = new ArrayList<ArrayList<String[]>>();
+			for(Map.Entry<String, UMLRelationship> entry : relationships.entrySet()) {
```
Comment (ryanhaas) [inline src/main/java/model/UMLClassManager.java]: This is likely what is causing your double print issue. I don't see why there would need to be a nested for loop here, unless I'm misreading this.

File: src/main/java/model/UMLClassManager.java [hunk 4]
```diff
@@ -413,34 +466,47 @@ public int removeRelationship(String srcClass, String type, String destClass) {
...
+			return result;
+		}
+	}
+	
 	/**
 	 * List the relationships the given class has
 	 * @param className
 	 * @return 'tuple' of [string of relationships, return code];
 	 */
-	public Object[] listRelationships(String className) {
```
Comment (ryanhaas) [inline src/main/java/model/UMLClassManager.java:421]: Not for future: Rewrite this method to return just an array of the relationships. Similar to the fields and methods.

**Review by ryanhaas — DISMISSED** [review]
Overall looks pretty good. Would definitely add more comments and possibly consider the recommendations. Looks good to merge to develop

File: src/main/java/controller/UMLController.java (hunk 2 above)
Comment (anthony81799) [reply src/main/java/controller/UMLController.java:35]: ConsoleView imports UMLComtroller and I call the print methods to list classes and relationships.

File: src/main/java/controller/GUIController.java (hunk 1 above)
Comment (anthony81799) [reply src/main/java/controller/GUIController.java:151]: Based on commit 6, if I delete the print methods it fails the build checks.

**Review by anthony81799 — COMMENTED** [review]
Looked at recommendations and made changes.

File: src/main/java/controller/GUIController.java (hunk 1 above)
Comment (ryanhaas) [reply src/main/java/controller/GUIController.java:151]: You likely would just have to cast the controller as a command controller or just set the controller to be of type CommandController from the beginning.

**Review by ryanhaas — APPROVED** [review]
Overall looks good for the most part to me

**Pull request merged**